modules/stanford_earth_capx/config/install/migrate_plus.migration.earth_capx_importer.yml

```diff
@@ -0,0 +1,150 @@
+uuid: 2200fcc9-747f-4e44-ae0d-b25826759bc3
+langcode: en
+status: true
+dependencies: {  }
+id: earth_capx_importer
+class: null
+field_plugin_method: null
+cck_plugin_method: null
+migration_tags: null
+migration_group: earth_capx
+label: 'Stanford Profiles (CAP) importer'
+source:
+  plugin: url
+  data_fetcher_plugin: http
+  data_parser_plugin: json
+  authentication:
+    plugin: oauth2
+    base_uri: 'https://authz.stanford.edu'
+    token_url: /oauth/token
+    grant_type: client_credentials
+    client_id: earth-sciences
```

Comment:
Copied from Slack:
I’ve removed the CAP API “client_secret” password value from the stanford_earth_helper/stanford_earth_capx module and have, for now, moved it into a configuration key stored in the private se3_blt repository, leaving behind a dummy value in the capx module. The key module makes it easy to take the real value and override the dummy value. Problem is that this is still not considered secure enough for a production system, though it is probably good enough for now.

Options for moving forward, for keys in general, but working specifically with the CAP API password for now...

1) Use a commercial key management service such as Lockr. Good: the key module supposedly integrates with some of these services; Bad: spending $ for yet another service.

2) Put the password in a file outside the docroot such as in an acquia “nobackup” directory, as described in https://docs.acquia.com/en/stable/acquia-cloud/manage/files/system-files/private/. Good: the key module makes it easy to point to this file and replace the dummy value in the migration with the file’s contents; no custom code required. Bad: the password is still stored in unencrypted text; we need to keep track of keys we store this way.

3) Use the Drupal encrypt module, along with the recommended Defuse PHP Encryption Library and Real AES Drupal module to encrypt/decrypt API keys using an encryption key which is stored outside the docroot. The encrypted API passwords can then either be stored safely in the configuration or also in files outside the docroot. Good: would be a good first step towards building a Stanford Drupal key-manager module; Bad: would require some custom coding to create a Key module plugin that combines encryption with authentication key type; for acquia systems, would also still require the master encryption key be stored in the nobackup directory as described above, and how secure is that, really?

Reply:
Replied back in slack.

1. Not a fan of yet another key manager system. Would rather see one of the other two options.

2. This looks like a decent low effort win. There are a number of certs, passwords, and other configuration already stored as unencrypted in non-web-accessible directories.  

3. This would be great and be the best for future growth and would love to see some effort put towards it but for level of effort vs wins I would say number 2 is the best choice for the project.